src/App.tsx

```diff
@@ -1,14 +1,70 @@
 /* eslint-disable max-len */
-import React from 'react';
+import { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [todoSelector, setTodoSelector] = useState('all');
+  const [todoFilter, setTodoFilter] = useState('');
+
+  useEffect(() => {
+    getTodos().then((receivedTodos) => {
+      if (receivedTodos.length > 0) {
+        const selectedTodos = receivedTodos.filter((todo) => {
+          const todoTitleIncludesFilter = todo.title
+            .toLowerCase()
+            .includes(todoFilter.toLowerCase());
+
+          switch (todoSelector) {
+            case 'active':
+              return !todo.completed && todoTitleIncludesFilter;
+            case 'completed':
+              return todo.completed && todoTitleIncludesFilter;
+            default:
+              return todoTitleIncludesFilter;
+          }
+        });
```

Comment:
Move this to visibleTodos variable using useMemo.
Set to state all todos from server